### ISSUE TITLE
unused_assignments lint false positive

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust#22630]
The following code produces an unused_assignment warning at line 5 ([playpen](http://is.gd/SeR8M6)):
```rust
fn main() {
    let mut done = false;
    while !done {
        done = true;
        continue;
    }
}
```

**Comment by steveklabnik** [comment]
Triage: no change

**Comment by dchammond** [comment]
Bumping: This issue has not yet been fixed.

**Comment by yuval-k** [comment]
Is this bug easy to fix? i would love to help, but have no idea where to start :)
Any pointers? even just the general direction of where to look would be great.

Thanks!

**Comment by oli-obk** [comment]
@yuval-k You can search the codebase for `UNUSED_ASSIGNMENTS`. There should be 2 or 3 mentions. The one that is surrounded by a `span_lint` call is where everything happens. From there you can go backwards to find out when that code is actually called.

**Comment by Amanieu** [comment]
Just hit this today, still an issue.
